aspnetcore/blazor/state-management.md

```diff
@@ -114,46 +201,273 @@ Caveats for using browser storage:
 
-## Third-party browser storage solutions
+::: moniker range=">= aspnetcore-5.0"
 
-Third-party NuGet packages provide APIs for working with `localStorage` and `sessionStorage`.
+## ASP.NET Core Protected Browser Storage
 
-It's worth considering choosing a package that transparently uses ASP.NET Core's [Data Protection](xref:security/data-protection/introduction). ASP.NET Core Data Protection encrypts stored data and reduces the potential risk of tampering with stored data. If JSON-serialized data is stored in plaintext, users can see the data using browser developer tools and also modify the stored data. Securing data isn't always a problem because the data might be trivial in nature. For example, reading or modifying the stored color of a UI element isn't a significant security risk to the user or the organization. Avoid allowing users to inspect or tamper with *sensitive data*.
+`Microsoft.AspNetCore.Components.Web.Extensions.ProtectedBrowserStorage` leverages [ASP.NET Core Data Protection](xref:security/data-protection/introduction) for [`localStorage`](https://developer.mozilla.org/docs/Web/API/Window/localStorage) and [`sessionStorage`](https://developer.mozilla.org/docs/Web/API/Window/sessionStorage).
 
-## Protected Browser Storage experimental package
+> [!NOTE]
+> Protected Browser Storage relies on ASP.NET Core Data Protection and is only supported for Blazor Server apps.
+
+### Save and load data within a component
+
+In any component that requires loading or saving data to browser storage, use the [`@inject`](xref:mvc/views/razor#inject) directive to inject an instance of either of the following:
 
-An example of a NuGet package that provides [Data Protection](xref:security/data-protection/introduction) for `localStorage` and `sessionStorage` is [`Microsoft.AspNetCore.ProtectedBrowserStorage`](https://www.nuget.org/packages/Microsoft.AspNetCore.ProtectedBrowserStorage).
+* `ProtectedLocalStorage`
+* `ProtectedSessionStorage`
+
+The choice depends on which browser storage location you wish to use. In the following example, `sessionStorage` is used:
+
+```razor
+@using Microsoft.AspNetCore.Components.Web.Extensions
+@inject ProtectedSessionStorage ProtectedSessionStore
+```
+
+The `@using` statement can be placed in the app's `_Imports.razor` file instead of in the component. Use of the `_Imports.razor` file makes the namespace available to larger segments of the app or the whole app.
```

Comment:
```suggestion
The `@using` directive can be placed in the app's `_Imports.razor` file instead of in the component. Use of the `_Imports.razor` file makes the namespace available to larger segments of the app or the whole app.
```